_references/component-descriptor-structure.md

```diff
@@ -0,0 +1,319 @@
+for component configuration. Every environment variable defined in the `component.json`
+file must have the following properties:
+
+| Property Name | Type     | Required | Description |
+| :------------ | :------: | :------: | :---------- |
+| required    | `boolean`|  | Specifies whether setting the environmental variable is required for the component to operate properly. If not present then the value is set to `false` by default. |
+| description | `string` |  | Description of the environment variable |
+
+The environment variable naming follows a standard UNIX/Linux Shell variable
+definitions. As a guidance, variable name must begin with an alphanumeric or alpha
+character, followed by one or more alphanumeric or underscore (`_`) characters.
+Avoid using reserved words like (`if`, `else`, `elif`, `do`, `done` ...). Here is
+an example of `envVars` object implemented in the [Salesforce Component](https://github.com/elasticio/salesforce-component/blob/master/component.json):
+
+```json
+"envVars": {
+    "SALESFORCE_KEY": {
+      "required": true,
+      "description": "Your Salesforce OAuth client key"
+    },
+    "SALESFORCE_SECRET": {
+      "required": true,
+      "description": "Your Salesforce OAuth client secret"
+    }
+  }
+```
+
+## Credentials Object
+
+The **Credentials Object** specifies how a user can grant a component the access
+to his protected resources using API. This is accomplished by defining fields
+used to gather authentication related data from the user.
+
+| Property Name | Type     | Required | Description |
+| :------------ | :------: | :------: | :---------- |
+| fields      | `object` | Yes | Object used to define input fields to gather authentication related data from the user. The keys of this object define field names, which must be unique. The values are definitions of the fields. |
+| [oauth1](#oauth1) | `object` |  | Specifies the details about OAuth v1.0 resources. Only used if a `OAuthFieldView` field is defined. |
+| [oauth2](#oauth2) | `object` |  | Specifies the details about OAuth v2.0 resources. Only used if a `OAuthFieldView` field is defined. |
+
+Let's explore an example from the [Petstore API (Node.js)](https://github.com/elasticio/petstore-component-nodejs/blob/master/component.json)
+component. The client authenticated with the Petstore API using an API key
+which is send via an HTTP header. That's why it is sufficient to define in
+the `credentials` object a single field to gather the API key from the user,
+as shown below.
+
+```json
+"credentials": {
+    "fields": {
+      "apiKey": {
+        "label": "API key",
+        "required": true,
+        "viewClass": "TextFieldWithNoteView",
+        "note": "Please use <b>elasticio</b> as API key"
+      }
+    }
+  }
+```
+
+In the example above the `apiKey` field is used to gather user's API key.
+The field is presented in the UI as a simple input field where a user must
```

Comment:
The view of the field is defined as `TextFieldWithNoteView` which tells the platform to present it in the UI as a simple input field where ... key. Please read more details on field types [here](LINK TO FIELDS paragraph).